### PR TITLE
remove extra db query in delete

### LINKED ISSUE #736: Registry server: remove extra db query in delete
In the actions layer, we query the database to check if the entity exists to return an error before performing a delete. This adds an unnecessary database call as we can check the return for RowsAffected == 0 on the target row.

### PR DESCRIPTION
**Pull request by theganyo**
Fixes #736

### CODE REVIEW

File: server/registry/internal/storage/delete.go [hunk 1]
```diff
@@ -303,8 +325,8 @@ func (c *Client) DeleteDeploymentRevision(ctx context.Context, name names.Deploy
 
 func (c *Client) DeleteArtifact(ctx context.Context, name names.Artifact) error {
 	for _, model := range []interface{}{
-		models.Blob{},
 		models.Artifact{},
+		models.Blob{},
```
Comment (timburks) [inline server/registry/internal/storage/delete.go:329]: Was this reordering functionally necessary? I was tempted to label it ["drive by fix"](http://tillenius.me/blog/2010/05/03/problems-with-drive-by-fixes/) but I'm wondering if there's an ordering sensitivity that we should note.

**Review by timburks — APPROVED** [review]
Nice improvement, thanks!

File: server/registry/internal/storage/delete.go (hunk 1 above)
Comment (theganyo) [reply server/registry/internal/storage/delete.go:329]: This is basically the same optimization: There's would be no point in calling the DB to delete blobs if the Artifact doesn't exist.

Comment (timburks) [reply server/registry/internal/storage/delete.go:329]: Makes sens, thanks!

**Review by seaneganx — APPROVED** [review]
